singer/src/main/java/com/pinterest/singer/utils/SingerUtils.java

```diff
@@ -91,6 +95,25 @@ public static String getHostname() {
     return hostName;
   }
 
+  /***
+   * Gradually builds substrings from hostname separated by dashes
+   * will return hostname if hostname does not contain dashes
+   *
```

Comment:
nit: An example would help to show what are considered hostname prefixes.

Also, considering the generic use case, should we somehow allow custom (configurable) separators (other than just `-`)?

Reply:
Yes, what do you think about setting a configurable regex pattern instead?

Reply:
That would be most flexible way of supporting this. Do you have an example?

Reply:
Yes for instance if you wanted hostname to be split by dots and dashes, we can have `optional string hostnamePrefixRegex = "[.-]";` then we split with this instead: `String [] splitHostname = HOSTNAME.split(singerConfig.getHostnamePrefixRegex());`